hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java

```diff
@@ -564,6 +564,9 @@ public synchronized void start(CertificateClient certClient)
       throws IOException {
     super.start(certClient);
     tokenRemoverThread = new Daemon(new ExpiredTokenRemover());
+    tokenRemoverThread.setName(
+        ozoneManager.getThreadNamePrefix() +
+            "ExpiredTokenRemover");
     tokenRemoverThread.start();
```

Comment:
Findbugs started complaining about:

```
M M IS: Inconsistent synchronization of org.apache.hadoop.ozone.security.OzoneDelegationTokenSecretManager.tokenRemoverThread; locked 42% of time  Unsynchronized access at OzoneDelegationTokenSecretManager.java:[line 579]
```

https://github.com/ArafatKhan2198/ozone/actions/runs/6557280070/job/17808617110#step:7:11

I think the warning is correct, though it should have complained about this earlier (the PR does not change synchronization).

Let's try fixing it:
 * change `public void stop()` to `public synchronized void stop()`
 * change `public void stopThreads()` to `private`

Reply:
It turns out `TestOzoneDelegationTokenSecretManager` is timing out due to this change.  It can be fxed by:

```diff
diff --git hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/OzoneSecretManager.java hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/OzoneSecretManager.java
index c99a13c489..601bdf0ea7 100644
--- hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/OzoneSecretManager.java
+++ hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/OzoneSecretManager.java
@@ -250,7 +250,7 @@ public Text getService() {
    *
    * @return true if secret mgr is running
    */
-  public synchronized boolean isRunning() {
+  public boolean isRunning() {
     return running;
   }
```

I hope it doesn't trigger another findbugs problem...

Reply:
Thanks for the help!
Ran it locally it worked just fine